pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -250,6 +251,7 @@ public int hashCode() {
     FLOAT,
     DOUBLE,
     BOOLEAN /* Stored as INT */,
+    BIG_DECIMAL /* Stored as BYTES */,
```

Comment:
(minor) Move it in front of BOOLEAN for consistency (Also keeping number types together)